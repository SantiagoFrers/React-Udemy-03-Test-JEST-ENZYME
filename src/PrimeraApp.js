import React from 'react'
import PropTypes from 'prop-types';

//const PrimeraApp = ( {saludo = 'Valor por defecto', adios} ) => {
//Tambien podemos enviar valores por defecto usando defaultProps, ver al final
const PrimeraApp = ( {
    saludo,
    adios
    } ) => {


        /* const objeto = {
            nombre: 'Santi',
            edad: 23
        }; */

        // Se puede imprimir strings, number, arrays
        // No se puede imprimir booleanos o objetos.
        //Para imprimir un objeto hay que JSON.stringify(objeto)

    return <>
        <h3>{ saludo } </h3>
        {/* <p> { JSON.stringify( objeto, null, 3) }</p> */}
        <p> { adios } </p>
    </>
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    adios: PropTypes.string
}

PrimeraApp.defaultProps = {
    adios: 'Soy un mensaje por defecto'
}

export default PrimeraApp;