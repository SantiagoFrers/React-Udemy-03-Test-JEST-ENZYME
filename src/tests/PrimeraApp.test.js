//Prueba sobre un componente de react

//import { render } from "@testing-library/react" // importamos esta libreria para poder renderizar componentes a evaluar
import { shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp" // importamos el componente

describe('Pruebas en <PrimeraApp />', ()=>{

//Esta pruena es con JEST con el ajuste en el archivo setupTest, no lo usaremos mas porque pasamos a ENZYME
/*     test('debe mostrar el mensaje Hola Pedro', () => {
        const saludo = 'Hola Pedro'
        const {getByText} = render( <PrimeraApp saludo={saludo}/>)

        expect(getByText(saludo)).toBeInDocument
        //expect(getByText(saludo + '!!' )).toBeInDocument - Prueba false
    }) */

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo = 'Hola Pedro'
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = 'Hola Pedro';
        const subtitulo = ' Soy un mensaje por defecto ';

        const wrapper = shallow(
            <PrimeraApp
                saludo = { saludo }
                subtitulo = { subtitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe(subtitulo)


    })

})