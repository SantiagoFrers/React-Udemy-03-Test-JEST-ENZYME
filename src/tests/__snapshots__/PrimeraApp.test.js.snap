// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h3>
    Hola Pedro
  </h3>
  <p>
     
    Soy un mensaje por defecto
     
  </p>
</Fragment>
`;
